nixos/modules/services/network-filesystems/moosefs.nix

```diff
@@ -66,40 +46,37 @@ let
 
 in {
   ###### interface
-
   options = {
     services.moosefs = {
       masterHost = lib.mkOption {
         type = lib.types.str;
         default = null;
-        description = "IP or DNS name of master host.";
+        description = "IP or DNS name of the MooseFS master server.";
       };
 
       runAsUser = lib.mkOption {
         type = lib.types.bool;
         default = true;
         example = true;
-        description = "Run daemons as user moosefs instead of root.";
+        description = "Run daemons as moosefs user instead of root for better security.";
       };
 
-      client.enable = lib.mkEnableOption "Moosefs client";
+      client.enable = lib.mkEnableOption ("MooseFS client");
 
       master = {
         enable = lib.mkOption {
           type = lib.types.bool;
           description = ''
-            Enable Moosefs master daemon.
-
-            You need to run `mfsmaster-init` on a freshly installed master server to
-            initialize the `DATA_PATH` directory.
+            Enable MooseFS master daemon. The service will automatically initialize on first run.
```

Comment:
This is quite dangerous and I would advise against automatically initializing the master's meta directory. Assume that you have accidentally not mounted the metadata directory on startup. This would result in your chunk servers starting to discard your data or at least end up an inconsistent state.
IMHO initialization should be a deliberate act. I wrote the init-script for exactly that reason.

Reply:
Hi,

I understand your concern, on my side all this must be automated and industrialized.
I will therefore see about adding additional security at this point.

Best Regards

Reply:
How about this: we leave init script in place, but add another option to the masters server's conifig: `autoInit => bool`. This can be `false` per default as safe guard against accidents. For your automated roll outs you can set `autoInit = true` and get what you need.